pkg/controller/jobs/job/job_controller.go

```diff
@@ -219,7 +220,8 @@ func (j *Job) RunWithPodSetsInfo(podSetsInfo []jobframework.PodSetInfo) {
 
 	if j.minPodsCount() != nil {
 		j.Spec.Parallelism = pointer.Int32(info.Count)
-		if j.syncCompletionWithParallelism() {
+		// only since kubernetes version 1.27 we can change parallelism and completions in tandem
+		if j.syncCompletionWithParallelism() && discoveryversion.AtLeast(discoveryversion.KubeVersion1_27) {
```

Comment:
Should we check the cluster version inside `syncCompletionWithparallelism`?

Reply:
It breaks the condition in webhook [here](https://github.com/kubernetes-sigs/kueue/blob/b2e8c9d0632c25c75b3ee8dfeecdce2bb6037464/pkg/controller/jobs/job/job_webhook.go#L165)